### DISCUSSION TITLE
New naming and overall concept for this repo

### DISCUSSION THREAD

**Discussion by aesedepece** [witnet/witnet-solidity-bridge#180] [Ideas]
Taking into account that:

- The software contained in this repo is now used on EVM-compatible chains that are not Ethereum.
- Some of the new targets are EVM-compatible but not strictly EVM, e.g. Boba is OVM.
- This repo will soon be better structured, with a clear separation between the contracts allowing the relay of requests and results between smart contract platforms and Witnet, and accessory contracts and interfaces.
- The current content of this repo is now available on NPM as `witnet-ethereum-bridge 0.2.0`.
- We have other solidity contracts in other repositories (e.g. `witnet-price-feed-examples`)
- Solidity monorepos are more convenient to manage, and quite standard today (see Chainlink, ADO, OpenZeppelin, etc.)

I'm proposing:

- Renaming this repo and its NPM package to `witnet-solidity`.
- Imports by end users would look like this: `import "witnet-solidity/contracts/UsingWitnet.sol"`.
- Move every relevant Solidity contract we have into this repo.
- Examples should have a directory for their own, something like `/contracts/examples/`.

**Comment by guidiaz** [comment]
I like the idea of changing `ethereum` to `solidity` within the repo's name. But naming it just as `witnet-solidity` would perhaps be too generic, as we do have other "solitidy" repos. So what about any of these other options?
- witnet-solidity-bridge
- witnet-solidity-board
- witnet-solidity-request-board

**Reply by aesedepece** [reply]
My point is that we may want not to have any more solidity repos and rather move everything here.

**Reply by aesedepece** [reply]
But maybe that's too big of a change, and given the latest changes on the price feeds example repository (which added a lot of boilerplate), my initial position of merging the examples into this repo are not a great idea anymore.

**Reply by aesedepece** [reply]
So I'm fine with `witnet-solidity-bridge`

**Reply by clbartoli** [reply]
I agree on having it renamed `witnet-solidity-bridge`. If the examples are the pricefeeds I also think is better to have them in a different repository so that we can update them or add new ones without affecting this repo. 
Anyway could be a good idea to add a link to the pricefeeds repository in the README of this repo, in case someone wants to see an example

**Comment by aesedepece** [comment]
I'd like to hear from @clbartoli.

Also summoning @mariocao just in case.

**Comment by aesedepece** [comment]
So, it looks like `witnet-solidity-bridge` be it!

**Reply by aesedepece** [reply]
Repo has been renamed. README renaming and documentation updating is due.
